### ISSUE TITLE
Segment fault when compiling LLVM with LLVM_ENABLE_MODULES=ON

### ISSUE THREAD

**Issue by junaire** [llvm/llvm-project#54726]
When I try to build LLVM with `-DLLVM_ENABLE_MODULES=ON`, everything is totally fine. But when I `checkout ` another branch, for example, my working branch `fix`, or switch back from `fix` to `main`,  it will report a "segment fault" and recompile some stuff.
Here are the screenshots:
![image](https://user-images.githubusercontent.com/77525145/161435053-535b536a-20b1-4065-91fd-9bd0e8c03851.png)
You can notice that there's a message "segment fault`, but the interesting thing is that it won't stop but will recompile more files. (The CXX objects need to be built increased from 225 to about 1000.

OS: Ubuntu 20.04 LTS
Compiler:
```
clang version 15.0.0 (https://github.com/llvm/llvm-project/ 8133778d3c8d481891143e21e0c5081a02a9889d)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/jun/opt/clang/bin
```
Linker: lld
Ninja version: 1.10.0
CMake config command:
```
cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo \
      -DCMAKE_C_COMPILER=clang \
      -DCMAKE_CXX_COMPILER=clang++ \
      -DCMAKE_EXPORT_COMPILE_COMMANDS=YES \
      -DLLVM_USE_LINKER=lld    \
      -DLLVM_TARGETS_TO_BUILD=X86\
      -DBUILD_SHARED_LIBS=ON   \
      -DLLVM_CCACHE_BUILD=ON   \
      -DLLVM_APPEND_VC_REV=OFF  \
      -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra" \
      -G"Ninja" ../llvm \
      -DLLVM_ENABLE_MODULES=ON
```

**Comment by junaire** [comment]
I'm not sure where the problem is, but it seems to be a ninja bug? Can anyone who has a powerful build bot confirm this? I mean try to use `make` instead of `ninja` to see what happens. If it is really a ninja bug, I think we have to remove `LLVM_ENABLE_MODULES` flag right?

**Comment by davidbolvansky** [comment]
Some time ago I saw such seg fault like yours.

Some files are autogenerated during LLVM build and when you switch branches they might be incompatible. You can enable debug mode in ninja to print building commands. If  the command fails, that failure confirms my first sentence.

**Comment by junaire** [comment]
> Some time ago I saw such seg fault like yours.
> 
> Some files are autogenerated during LLVM build and when you switch branches they might be incompatible. You can enable debug mode in ninja to print building commands. If  the command fails, that failure confirms my first sentence.

yeah I agree. I have seen that it complains that some `.pcm` files are not matched and try to regenerate after the `segment fault` message.
